python/lsst/alert/stream/alertProducer.py

```diff
@@ -12,34 +15,39 @@ class AlertProducer(object):
     ----------
     topic : `str`
         The name of the topic stream for writing.
-    schema : Avro schema
-        The writer Avro schema for encoding data. Optional.
     **kwargs
         Keyword arguments for configuring confluent_kafka.Producer().
     """
 
-    def __init__(self, topic, schema_files=None, **kwargs):
+    def __init__(self, topic, **kwargs):
         self.producer = confluent_kafka.Producer(**kwargs)
         self.topic = topic
-        if schema_files is not None:
-            self.alert_schema = avroUtils.combineSchemas(schema_files)
 
-    def send(self, data, encode=False):
+    def send(self, schema, data):
         """Sends a message to Kafka stream.
 
+        The message is encoded following the `Confluent Wire Format`_. Thus:
+
+        Byte 0:     ``0``.
+        Byte 1-4:   A 4-byte schema ID.
+        Byte 5-...: The packet data, Avro encoded following `schema`.
+
+        .. _Confluent Wire Format: https://docs.confluent.io/current/schema-registry/docs/serializer-formatter.html#wire-format
+
         Parameters
         ----------
-        data : message content
-            Data containing message content.  If encode is True, expects JSON.
-        encode : `boolean`
-            If True, encodes data to Avro format. If False, sends data raw.
+        schema : `lsst.alert.packet.Schema`
+            The Avro schema for encoding data. Although one is not required to
+            construct the producer, you can't begin sending data until one has
+            been supplied.
+        data : `dict`
+            Message content. Must comply with the `schema`.
         """
-        if encode is True:
-            avro_bytes = avroUtils.writeAvroData(data, self.alert_schema)
-            raw_bytes = avro_bytes.getvalue()
-            self.producer.produce(self.topic, raw_bytes)
-        else:
-            self.producer.produce(self.topic, data)
+        outgoing_bytes = BytesIO()
+        outgoing_bytes.write(struct.pack("!b", 0))
+        outgoing_bytes.write(struct.pack("!I", SchemaRegistry.calculate_id(schema)))
```

Comment:
This appears to be a bug--we need to instantiate a concrete `SchemaRegistry` in order to call `calculate_id`.

Otherwise I get:
```Traceback (most recent call last):
  File "bin/sendAlertStream.py", line 47, in schedule_delays
    yield from asyncio.ensure_future(delay(wait_time, function, arg))
  File "bin/sendAlertStream.py", line 27, in delay
    return function(*args)
  File "bin/sendAlertStream.py", line 80, in send_visit
    streamProducer.send(schema, record)
  File "/home/alert_stream/python/lsst/alert/stream/alertProducer.py", line 48, in send
    outgoing_bytes.write(struct.pack("!I", SchemaRegistry.calculate_id(schema)))
AttributeError: type object 'SchemaRegistry' has no attribute 'calculate_id'
```

Reply:
Hmm, weird, we _shouldn't_ need to instantiate it, because it's a [static method](https://github.com/lsst-dm/sample-avro-alert/blob/tickets/DM-17549/python/lsst/alert/packet/schemaRegistry.py#L79).

My guess is actually that you are running in Docker (following the instructions), but have an old version of sample-avro-alert cached on your system. Could you try running `docker build --no-cache -t "alert_stream" .` and see if that fixes the problem?

Reply:
I ultimately had to wipe out my entire Docker cache, but it does seem to work now.  `monitorStream.py` doesn't seem to behave quite the way I remember, but I can't really make sense of it from the code, and it's not essential.